store/keyboard/index.tsx

```diff
@@ -0,0 +1,55 @@
+import {
+  createContext,
+  FC,
+  KeyboardEvent,
+  ReactElement,
+  useContext,
+  useState,
+} from "react";
+import { ALT_KEY } from "@/constants/AppConstants";
+import { KEYBOARD_CONTEXT } from "@/types/member_store";
+
+const KeyboardContext = createContext<KEYBOARD_CONTEXT>({isOptionKeyPressed : false});
+
+function isOptionKey(e: KeyboardEvent) {
+  return e.key === ALT_KEY;
+}
+
+export const KeyboardProvider: FC<{ children: ReactElement }> = ({
+  children,
+}): ReactElement => {
+  const [isOptionKeyPressed, setIsOptionKeyPressed] = useState(false);
+
+  const initialValue = {
+    isOptionKeyPressed,
+  };
+
+  return (
+    <KeyboardContext.Provider value={initialValue}>
+      <div
+        role="button"
+        tabIndex={0}
+        onKeyDown={(e) => {
+          if (isOptionKey(e)) {
```

Comment:
i see a repeating pattern in `onKeyDown `and `onKeyup ` we can take it out as a function and let the function handle the cases inside rather than having 2 almost identical blocks where we are just setting boolean.